### PR TITLE
Improved lookup mechanism for .md files

### PR DESCRIPTION
**Pull request by Bee-lee**
# Description

For https://issues.redhat.com/browse/CCXDEV-4454

- First tries to look for all .md files in error key directory
- Then tries to look for them in plugin directory
- Copies the contents of files in plugin directory to the error keys, if error keys are empty
- Checks if `generic.md` and `reason.md` are present for all error keys

It's possible to have all .md files on rule plugin level or on error key level. On plugin level, only `plugin.yaml` is mandatory, on error key level, only the `metadata.yaml` is mandatory. `generic.md` and `reason.md` have to be present on atleast one level.

Fixes CCXDEV-4454

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update
- Refactor (refactoring code, removing useless files)
- Unit tests (no changes in the code)
- Documentation update

## Testing steps
`make before_commit`, testing alongside smart-proxy seemed OK without any changes, need to look more

## Checklist
* [x] `make before_commit` passes
* [ ] updated documentation wherever necessary
* [x] added or modified tests if necessary
* [ ] updated schemas and validators in [insights-data-schemas](https://github.com/RedHatInsights/insights-data-schemas) in case of input/output change

### CODE REVIEW

File: content/content.go [hunk 1]
```diff
@@ -33,6 +34,17 @@ import (
 const (
 	separator          = "------------------------------------------------------------"
 	directoryAttribute = "directory"
+
+	// PluginYAML represents the filename of rule's plugin specification
+	PluginYAML = "plugin.yaml"
+	// MetadataYAML represents the filename of rule error key's metadata
+	MetadataYAML = "metadata.yaml"
+
+	GenericMarkdown    = "generic.md"
```
Comment (tisnik) [inline content/content.go:43]: would it be possible please to put some comment about these file names?

File: content/content.go [hunk 2]
```diff
@@ -53,6 +65,24 @@ type (
...
+var (
+	// MandatoryRuleWideContentFiles are mandatory files that MUST be on rule plugin or error key level
+	MandatoryRuleWideContentFiles = []string{GenericMarkdown, ReasonMarkdown}
+
+	// SharedContentFiles MAY be on either the plugin level or the error key level
+	SharedContentFiles = []string{GenericMarkdown, ReasonMarkdown, SummaryMarkdown, ResolutionMarkdown, MoreInfoMarkdown}
+
+	// RulePluginMandatoryContentFiles are mandatory on the plugin level
+	RulePluginMandatoryContentFiles = []string{PluginYAML}
+	// RulePluginContentFiles are all files to look for on rule plugin level
```
Comment (tisnik) [inline content/content.go:77]: nitpick -prob. empty line make it more readable

File: content/content.go [hunk 3]
```diff
@@ -72,31 +102,137 @@ func readFilesIntoFileContent(baseDir string, filelist []string) (map[string][]b
...
+			if errorKey.MoreInfo == "" {
+				ek.MoreInfo = ruleContent.MoreInfo
+			}
+		default:
+			log.Error().Msgf("Behaviour for copying contents of file `%v` to error keys is not defined.", filename)
+			break
+		}
+
+		errorKeys[i] = ek
+	}
```
Comment (tisnik) [inline content/content.go:171]: interesting solution! Thank you

File: content/content.go [hunk 4]
```diff
@@ -72,31 +102,137 @@ func readFilesIntoFileContent(baseDir string, filelist []string) (map[string][]b
...
+
+}
+
+func isPluginFileMandatory(fname string) bool {
+	for _, v := range RulePluginMandatoryContentFiles {
```
Comment (tisnik) [inline content/content.go:176]: dunno if `StringInSlice` from `utils` package will be useful there?

File: content/content.go [hunk 5]
```diff
@@ -72,31 +102,137 @@ func readFilesIntoFileContent(baseDir string, filelist []string) (map[string][]b
...
+		if fname == v {
+			return true
+		}
+	}
+	return false
+}
+
+func isErrorKeyFileMandatory(fname string) bool {
+	for _, v := range ErrorKeyMandatoryContentFiles {
```
Comment (tisnik) [inline content/content.go:185]: dtto

File: gocyclo.sh [hunk 6]
```diff
@@ -20,4 +20,4 @@ then
     GO111MODULE=off go get github.com/fzipp/gocyclo/cmd/gocyclo
 fi
 
-gocyclo -over 9 -avg .
+gocyclo -over 13 -avg .
```
Comment (tisnik) [inline gocyclo.sh:23]: bitten by it, right? :)

**Review by joselsegura — APPROVED** [review]
It looks very nice. TYVM

File: content/content.go (hunk 4 above)
Comment (Bee-lee) [reply content/content.go:176]: ah, of course! thanks

File: content/content.go (hunk 2 above)
Comment (Bee-lee) [reply content/content.go:82]: done, thank you

File: content/content.go (hunk 1 above)
Comment (Bee-lee) [reply content/content.go]: done! thank you

File: content/content.go (hunk 4 above)
Comment (Bee-lee) [reply content/content.go]: done

File: content/content.go (hunk 5 above)
Comment (Bee-lee) [reply content/content.go]: done, thanks!

**Pull request merged**